services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/ApplicationConfig.java

```diff
@@ -473,7 +473,7 @@ public AmqpEndpoint credentialsAmqpEndpoint() {
         // need to use backend service because the Credentials service's "get" operation
         // supports auto-provisioning and thus needs to be able to create a device on the fly
         final MongoDbBasedDeviceBackend service = new MongoDbBasedDeviceBackend(registrationService(),
-                credentialsService());
+                credentialsService(), tenantInformationService());
```

Comment:
this will create yet another TenantService instance in addition to the ones created by *registrationService()* and *tenantHttpEndpoint()* and *tenantAmqpEndpoint()* ... We really need to get rid of the auto wiring and instead start assembling the objects manually so that we do not create all these duplicate service instances ....

Reply:
Do you suggest to use `new AutowiredTenantInformationService()` instead of `tenantInformationService()`?

Reply:
No, I suggest to implement #2110 and stop using auto-wiring for instantiating the beans because it doesn't give us the kind of control over the instances being used that we need ... But that should be a next step.